src/opsinputs/opsinputs_fill_mod.F90

```diff
@@ -1057,52 +1060,76 @@ subroutine opsinputs_fill_fillreal2d_norecords( &
 integer                                         :: iChannel
 integer                                         :: offset
 integer                                         :: numchans
-logical                                         :: localUseActualChans
+integer                                         :: offsetsize
+integer                                         :: arrayindex
+logical                                         :: compressChannels
 
 ! Body:
 
 MissingDouble = missing_value(0.0_c_double)
 
+compressChannels = .true.
+if (present(compressVarChannels)) then
+  compressChannels = compressVarChannels
+end if
+
 JediVarNamesWithChannels = opsinputs_fill_varnames_with_channels(JediVarName, Channels)
 
 !take into account offsetting of 2nd dimension if required
 !designed to be used to pack where multiple satellite instruments expected
 !e.g. HIRS in ATOVS stream
-offset = 0
+
 numchans = size(JediVarNamesWithChannels)
-if (present(OffsetChans)) then
-  offset = OffsetChans % channel_offset
-  if (OffsetChans % size_of_varobs_array > 0) &
-    numchans = OffsetChans % size_of_varobs_array
-end if
 
-!Setup for channels needing to match array index
-localUseActualChans = .false.
-if (present(useActualChans)) then
-    localUseActualChans = useActualChans
+if (present(sizeVarObs)) then
+  if (sizeVarObs > 0) then
+    numchans = sizeVarObs
+  end if
 end if
 
 if (obsspace_has(ObsSpace, JediVarGroup, JediVarNamesWithChannels(1))) then
   ! Allocate OPS data structures
   call Ops_Alloc(Hdr, OpsVarName, NumObs, Real2, &
                  num_levels = int(numchans, kind=integer64))
   do iChannel = 1, size(JediVarNamesWithChannels)
-    ! Retrieve data from JEDI
     call obsspace_get_db(ObsSpace, JediVarGroup, JediVarNamesWithChannels(iChannel), VarValue)
-
-    ! Fill the OPS data structures
-    if (localUseActualChans) then
-      where (VarValue /= MissingDouble)
-        Real2(:, Channels(iChannel)) = VarValue
-      end where
+    arrayindex = iChannel
+
+    ! if VAR channels have been assigned then jopa channels will be mapped to these var channels
+    ! Set up the size of the array, if channels are being pushed together an offset between
+    ! the var and jopa channel numbers is added onto the size of the array.
+    ! If not compressed the positions in the array are based on the actual channel number.
+
+    if (present(varChannels)) then
+      if (size(varChannels) > 0) then
+        if (iChannel <= size(varChannels)) then
+          if (compressChannels) then
+            offsetsize = abs(varChannels(1) - channels(1))
```

Comment:
I am not really understanding what is happening in this section. 
Also, this assumes a constant offset - is this true? What if we have 
```
channels: 1, 3
varChannels: 2, 5
```

Reply:
This is setting the index positioning. The constant offset is replicating how the previous offset_channels worked, this included a number of empty spaces at the start the output array. So in the example you have provided it would set an offset of 1 so there would be an empty space at the start of the array and then channel2 and 5 in the subsequent two spaces.